src/common.jl

```diff
@@ -336,6 +333,15 @@ function assign_reference_buses(
     return bus_groups
 end
 
+function assign_reference_buses!(
```

Comment:
Ideally we'd get a test case that demonstrates what breaks without this new layer of indirection — if too complicated to do now but potentially feasible in the future, add a `# TODO` or GitHub Issue